Sources/SwiftRedis/RedisResp.swift

```diff
@@ -207,30 +216,28 @@ class RedisResp {
     // Mark: Parser helper functions
 
     private func compare(_ buffer: inout Data, at offset: Int, with: Data) throws -> (Bool, Int) {
-        while  offset+with.count >= buffer.count {
+        // use offset+1 instead of offset+with.count as with.count is always == 1
```

Comment:
Is this no reducing the genericity of this function?
If this has no major impact, I would rather keep it as it was, otherwise, better to create another function with a different name and doc.